src/main/java/tech/allegro/wildsnake/application/api/ProductsController.java

```diff
@@ -0,0 +1,49 @@
+package tech.allegro.wildsnake.application.api;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import tech.allegro.wildsnake.product.model.Product;
+import tech.allegro.wildsnake.product.repository.ProductRepository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Created by michalolszewki on 2015-12-06.
+ */
+@RestController
+@RequestMapping(value = "api/v1/products")
+public class ProductsController {
+    @Autowired
+    ProductRepository productRepository;
+
+    private Iterable<Product> productList = new ArrayList<>();
```

Comment:
Do you really need private field? The field is mutable so can be changed by every method of this class. Very unpredictable. Even if "Law of Demeter" is not broken but I would suggest create local variable.